### PR TITLE
Allow virtualenvs to find Python bindings provided by deps

### PR DESCRIPTION
**Pull request by tdsmith**
cf #603

### CODE REVIEW

File: Library/Homebrew/language/python.rb [hunk 1]
```diff
@@ -122,6 +122,17 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         ENV.refurbish_args
         venv = Virtualenv.new formula, venv_root, python
         venv.create
+
+        # Find any Python bindings provided by dependencies
+        deps = formula.deps.select do |d|
+          !d.build? && (d.required? || formula.build.with?(d))
```
Comment (ilovezfs) [inline Library/Homebrew/language/python.rb]: does this ignore optional & required?

Comment (tdsmith) [inline Library/Homebrew/language/python.rb]: `formula.build.with?(d)` handles optional and recommended.

File: Library/Homebrew/language/python.rb [hunk 2]
```diff
@@ -122,6 +122,17 @@ def virtualenv_create(venv_root, python = "python", formula = self)
...
+        end
+        xy = Language::Python.major_minor_version python
+        pth_contents = deps.map do |d|
+          "import site; site.addsitedir('#{Formula[d.name].opt_lib}/python#{xy}/site-packages')\n"
+        end
+        (venv_root/"lib/python#{xy}/site-packages/homebrew_deps.pth").write pth_contents.join
```
Comment (ilovezfs) [inline Library/Homebrew/language/python.rb]: can this be named anything?

File: Library/Homebrew/language/python.rb [hunk 3]
```diff
@@ -122,6 +122,17 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         ENV.refurbish_args
         venv = Virtualenv.new formula, venv_root, python
         venv.create
+
+        # Find any Python bindings provided by dependencies
+        deps = formula.deps.select do |d|
```
Comment (ilovezfs) [inline Library/Homebrew/language/python.rb]: should this be `recursive_dependencies`?

File: Library/Homebrew/language/python.rb (hunk 2 above)
Comment (tdsmith) [inline Library/Homebrew/language/python.rb]: Yes.

File: Library/Homebrew/language/python.rb (hunk 3 above)
Comment (tdsmith) [inline Library/Homebrew/language/python.rb]: It could be; I think either behavior is probably okay.

**Comment by MikeMcQuaid** [comment]
If it fixes it: seems fine to me 👍

**Comment by DomT4** [comment]
Are you going to throw me in the sea if I tell you this doesn't _quite_ work on my example in the Issue because of the way a formula is packaged? 🙈.

```python
[ERROR   ] 02:19:21 ui:171 No module named gtk
Traceback (most recent call last):
  File "/usr/local/Cellar/deluge2/1.3.13/libexec/lib/python2.7/site-packages/deluge/ui/ui.py", line 152, in __init__
    from deluge.ui.gtkui.gtkui import GtkUI
  File "/usr/local/Cellar/deluge2/1.3.13/libexec/lib/python2.7/site-packages/deluge/ui/gtkui/__init__.py", line 1, in <module>
    from gtkui import start
  File "/usr/local/Cellar/deluge2/1.3.13/libexec/lib/python2.7/site-packages/deluge/ui/gtkui/gtkui.py", line 50, in <module>
    reactor = gtk2reactor.install()
  File "/usr/local/Cellar/deluge2/1.3.13/libexec/lib/python2.7/site-packages/twisted/internet/gtk2reactor.py", line 99, in install
    reactor = Gtk2Reactor(useGtk)
  File "/usr/local/Cellar/deluge2/1.3.13/libexec/lib/python2.7/site-packages/twisted/internet/gtk2reactor.py", line 71, in __init__
    import gtk as _gtk
ImportError: No module named gtk
```
The problem here being that our `pygtk` formula's `gtk` module is packaged in:

```
/usr/local/opt/pygtk/lib/python2.7/site-packages/gtk-2.0
```
Rather than:
```
/usr/local/opt/pygtk/lib/python2.7/site-packages
```

Not sure if that's an upstream issue or something problematic to this PR, but thought I'd mention it. Everything else about this is 💯 on solving the other issues encountered.

**Comment by tdsmith** [comment]
> Are you going to throw me in the sea

the North Atlantic, post-haste!

The version that operates on recursive dependencies *should* work here since the relevant file is provided by `pygobject` and pygtk depends_on pygobject; is that not happening?

**Comment by tdsmith** [comment]
Oh: no, it's not happening, because the .pth file pygobject installs is assuming that it's read from the same tree as pygtk, when they're actually in separate cellars. pygobject's .pth file should be rewritten.

**Comment by DomT4** [comment]
> the North Atlantic, post-haste!

Any chance we can go for a warmer sea?

> is that not happening?

I ran `brew pull` on this PR & then `brew install ~/Downloads/deluge2.rb`, and then executing a simple `deluge`, so I guess not?

**Comment by DomT4** [comment]
Thanks Tim ❤️

**Pull request merged**